### PR TITLE
Feature 117

### PR DESCRIPTION
**Pull request by f18m**
This is the initial PR related to the #117 .
Work-in-progress is about following areas:

1) bash editor preference page:
   - it would be nice to have the page resizeable so that if a long external tool command line is written the user is able to resize the window to view the entire command line; currently BashEditor pref page does not resize. I tried to fix the issue but failed
  - validation of external tool... maybe we should at least check the user has a) put non-empty string, b) used the $filename placeholder

2) reformatting on save:
  - current code to relocate the caret after refreshing content may be improved; it's the result of several attempts though

### CODE REVIEW

**Comment by de-jcup** [comment]
@f18m : Could you please change your pull request target to branch [feature-117](https://github.com/de-jcup/eclipse-bash-editor/tree/feature-117) ? As long as the feature is not 100% ready for next build I would like to have it not at master branch.

**Comment by f18m** [comment]
Hi @de-jcup,
I think the PR is already targeting that branch... at the top of this page I see
 "f18m wants to merge 8 commits into de-jcup:feature-117 from f18m:feature-117"

...

**Comment by f18m** [comment]
Btw I have: 
 - merged with master branch of eclipse-bash-editor after new release (nice job btw!)
 - fixed an issue with Eclipse marking as dirty the document after the reload.

At this point I will wait for a first review before proceeding, just to make sure we're aligned :)

Also if you have any suggestions on how to fix the problems 1 and 2 listed in my first comment that would be great... I also have no good idea on how to implement the validation of the text control for the external tool... I guess I can take inspiration by the source code of other plugins...

File: basheditor-plugin/src/main/java-eclipse/de/jcup/basheditor/preferences/BashEditorPreferencePage.java [hunk 1]
```diff
@@ -222,11 +226,10 @@ public void widgetSelected(SelectionEvent e) {
 		codeAssistGroupLayoutData.widthHint=400;
 		
 		Group codeAssistGroup = new Group(appearanceComposite,SWT.NONE);
-		codeAssistGroup.setText("Code assistence");
+		codeAssistGroup.setText("Code assistance");
```
Comment (de-jcup) [inline basheditor-plugin/src/main/java-eclipse/de/jcup/basheditor/preferences/BashEditorPreferencePage.java]: Thanks for removing the typo...

File: basheditor-plugin/src/main/java-eclipse/de/jcup/basheditor/BashEditor.java [hunk 2]
```diff
@@ -774,27 +774,32 @@ protected void performSave(boolean overwrite, IProgressMonitor progressMonitor)
...
+				textWidget.setCaretOffset(lastCaretPos);
+				BashEditorUtil.logInfo("Moved the caret at: " + textWidget.getCaretOffset());
 				textWidget.showSelection();
+
+				// PROBLEM: IN THIS WAY THE DOCUMENT IS MARKED AS "DIRTY" BY ECLIPSE AND THUS
+				// AN ASTERISK WILL APPEAR IN THAT DOCUMENT TAB AND ECLIPSE WILL ASK TO RELOAD IT
+				// SINCE IT WAS CHANGED ON DISK!!!
+
+				// hackish way to remove the "dirty" status from the document: re-save with overwrite=true
+				super.performSave(true, progressMonitor);
```
Comment (de-jcup) [inline basheditor-plugin/src/main/java-eclipse/de/jcup/basheditor/BashEditor.java]: I think this is okay . Maybe https://wiki.eclipse.org/FAQ_When_should_I_use_refreshLocal%3F would be another option which could save much code here.

**Comment by de-jcup** [comment]
You are right the merge will not go to master but feature-branch. 

I will merge the PR and review, test and give feedback then.

**Pull request merged**